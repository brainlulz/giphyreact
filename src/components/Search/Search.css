.Search {
  grid-column: 2/3;
  grid-row: 2/2;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 3rem;
  color: #FFBD39;
}

.SearchBar {
  align-self: center;
  justify-self: center;
}

input {
  outline: none;
  display: block;
  border: 5px #FFBD39 solid;
  background: transparent;
  width: 15.5rem;
  height: 2.5rem;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  position: relative;
  overflow: auto;
  font-family: 'SubZER0';
  font-size:1.5rem;
  color: #FFBD39;
}

.SubmitBtn {
  background: transparent;
  display: inline-block;
  background-image: url('https://image.noelshack.com/fichiers/2018/09/1/1519661192-export-3.png');
  background-size: 100%;
  margin: 0.8rem 0.3rem 0.5rem 0.5rem;
  width: 1rem;
  height: 1rem;
  background-repeat: no-repeat;
  position: absolute;
  border: none;
  z-index: 1;
}