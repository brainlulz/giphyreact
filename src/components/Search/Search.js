import React from "react";
import { Link } from "react-router-dom";
import "./Search.css";

const Search = ({ onSubmit, onChange, value }) => (
  <div className="Search">
    <div className="SearchBar">
      <form action="" id="form" onSubmit={onSubmit}>
        <Link to={{ pathname: "/search/", search: `?q=${value}` }}>
          <button type="submit" className="SubmitBtn" />
        </Link>
        <input type="search" onChange={onChange} />
      </form>
    </div>
  </div>
);

export default Search;
