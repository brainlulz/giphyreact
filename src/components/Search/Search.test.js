import React from 'react';
import { shallow } from 'enzyme';
import Search from './index';

describe('Search', () => {
  it('renders the Search component', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
});