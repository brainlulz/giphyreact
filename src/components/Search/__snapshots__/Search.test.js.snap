// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders the Search component 1`] = `
<div
  className="Search"
>
  <div
    className="SearchBar"
  >
    <form
      id="form"
      onSubmit={[Function]}
    >
      <button
        className="SubmitBtn"
        type="submit"
      />
      <input
        onChange={[Function]}
        type="search"
      />
    </form>
  </div>
</div>
`;
